src/main/java/org/apache/mesos/scheduler/DefaultScheduler.java

```diff
@@ -279,7 +279,8 @@ public void run() {
                 // UNRESERVE and DESTROY (in the case of persistent volumes) Operations.
                 ResourceCleanerScheduler cleanerScheduler = getCleanerScheduler();
                 if (cleanerScheduler != null) {
-                    acceptedOffers.addAll(getCleanerScheduler().resourceOffers(driver, offers));
+                    unacceptedOffers = filterAcceptedOffers(unacceptedOffers, acceptedOffers);
```

Comment:
Why do we need to filter Offers here?  The `ResourceCleanerScheduler` only `UNRESERVE`s and `DESTROY`s unexpected resources and volumes.  I think I've considered doing this in the past but couldn't convince myself that it mattered whether the Offers had been accepted or not.